src/core/imports/cuda.cpp

```diff
@@ -34,47 +38,14 @@ void InitializeCUDA(int argc, char* argv[])
 
         // Get local rank (rank within compute node)
         int localRank = 0;
-        char* env=nullptr;
+        char* env = nullptr;
         if (!env) { env = std::getenv("SLURM_LOCALID"); }
         if (!env) { env = std::getenv("MV2_COMM_WORLD_LOCAL_RANK"); }
         if (!env) { env = std::getenv("OMPI_COMM_WORLD_LOCAL_RANK"); }
         if (env) { localRank = std::atoi(env); }
 
         // Try assigning GPUs to local ranks in round-robin fashion
         device = localRank % numDevices;
-
-        // Check GPU compute mode
-        EL_FORCE_CHECK_CUDA_NOSYNC(
-            cudaGetDeviceProperties(&deviceProp, device));
-        switch (deviceProp.computeMode)
-        {
-        case cudaComputeModeExclusive:
-        case cudaComputeModeExclusiveProcess:
-            if (localRank >= numDevices)
-            {
-                cudaDeviceReset();
-                RuntimeError("CUDA device ",device," has a compute mode "
-                             "that does not permit sharing amongst ",
-                             "multiple MPI ranks");
-            }
-            else
-            {
-                // Let CUDA handle GPU assignments
-                EL_FORCE_CHECK_CUDA_NOSYNC(cudaGetDevice(&device));
-            }
-            break;
-        }
-
-    }
-
-    // Check GPU compute mode
-    EL_FORCE_CHECK_CUDA_NOSYNC(
-        cudaGetDeviceProperties(&deviceProp, device));
-    if (deviceProp.computeMode == cudaComputeModeProhibited)
-    {
-        cudaDeviceReset();
-        RuntimeError("CUDA Device ",device,
-                     " is set with ComputeModeProhibited");
     }
```

Comment:
I understand the need for this, and I agree that it is good to avoid the unnecessary memory allocation on the first GPU, but can we make a note in a comment in the code about the error checking that has been lost.  That way if someone figures out a good way to do it with NVML we have a reminder of what we lost.